pkg/networkmanager/firewall-client.js

```diff
@@ -327,4 +341,70 @@ firewall.removeServiceFromZones = (zones, service) => {
     return Promise.all(zones.map(z => firewall.removeService(z, service)));
 };
 
+firewall.activateZone = (zone, interfaces, sources) => {
+    let p = Promise.resolve();
+    for (let i of interfaces)
+        p = p.then(() => firewalld_dbus.call('/org/fedoraproject/FirewallD1',
+                                             'org.fedoraproject.FirewallD1.zone',
+                                             'addInterface', [zone, i]));
```

Comment:
This seems to be some fancy way of a dynamic chained promise? Why can't this be done through `cockpit.all()` instead? This at least deserves a comment.